src/yum-plugins/enabled_repos_upload.py

```diff
@@ -1,108 +1,38 @@
-import sys
 import os
-try:
-    import json
-except ImportError:
-    import simplejson as json
-
 from yum import YumBase
-sys.path.append('/usr/share/rhsm')
-
 from yum.plugins import TYPE_CORE, TYPE_INTERACTIVE
 
-try:
-    from subscription_manager.identity import ConsumerIdentity
-except ImportError:
-    from subscription_manager.certlib import ConsumerIdentity
-
-from rhsm.connection import UEPConnection, RemoteServerException, GoneException
+from katello.repos import upload_enabled_repos_report
 
-try:
-    from subscription_manager.injectioninit import init_dep_injection
-    init_dep_injection()
-except ImportError:
-    pass
+from logging import Logger
 
 requires_api_version = '2.3'
 plugin_type = (TYPE_CORE, TYPE_INTERACTIVE)
 
-def upload_enabled_repos_report():
-    path = '/etc/yum.repos.d/redhat.repo'
-    uep = UEP()
-    report = EnabledReport(path)
-    content = report.content
-    consumer_id = lookup_consumer_id()
-    if consumer_id is None:
-        error_message('Cannot upload enabled repos report, is this client registered?')
-    else:
-        cache = EnabledRepoCache(consumer_id, content)
-        if not cache.is_valid():
-            uep.report_enabled(consumer_id, content)
-            cache.save()
-
-def error_message(msg):
-    sys.stderr.write(msg + "\n")
-
-def lookup_consumer_id():
-    try:
-        certificate = ConsumerIdentity.read()
-        return certificate.getConsumerId()
-    except IOError:
-        return None
-
-class EnabledRepoCache:
-    CACHE_FILE = '/var/cache/katello-agent/enabled_repos.json'
-    def __init__(self, consumer_id, content):
-        self.consumer_id = consumer_id
-        self.content = content
-
-    @staticmethod
-    def remove_cache():
-        try:
-            os.remove(EnabledRepoCache.CACHE_FILE)
-        except OSError:
-            pass
-
-    def is_valid(self):
-        if not os.path.isfile(self.CACHE_FILE):
-            return False
-        file = open(self.CACHE_FILE)
-        try:
-            return self.data() == json.loads(file.read())
-        except ValueError:
-            return False
-
-    def data(self):
-        return {self.consumer_id: self.content}
-
-    def save(self):
-        file = open(self.CACHE_FILE, 'w')
-        file.write(json.dumps(self.data()))
-        file.close()
-
-class UEP(UEPConnection):
+class Yum(YumBase):
     """
-    Represents the UEP.
+    Provides custom configured yum object.
     """
 
-    def __init__(self):
-        key = ConsumerIdentity.keypath()
-        cert = ConsumerIdentity.certpath()
-        UEPConnection.__init__(self, key_file=key, cert_file=cert)
+    def cleanLoggers(self):
```

Comment:
the convention for functions would be `clean_loggers` unless this is overriding something.

Reply:
'tis an override